### PR TITLE
refactored retrieval of template

### PR DESCRIPTION
**Pull request by goern**
refactored retrieval of template to OpenShift._get_template(label_selector)

Signed-off-by: Christoph Görn <goern@redhat.com>

### CODE REVIEW

**Comment by goern** [comment]
do we have to roll a new release now?

File: thoth/common/openshift.py [hunk 1]
```diff
@@ -382,7 +384,7 @@ def _status_report(status):
         for key, value in status[state].items():
             if key == "containerID":
                 value = (
-                    value[len("docker://"):]
+                    value[len("docker://") :]
```
Comment (fridex) [inline thoth/common/openshift.py]: Coala will complain here.

File: thoth/common/openshift.py [hunk 2]
```diff
@@ -472,6 +474,19 @@ def get_jobs(self, label_selector: str, namespace: str = None) -> dict:
 
         return response
 
+    def _get_template(self, _label_selector: str) -> dict:
+        """Get template from infra namespace, use label_selector to identify which template to get."""
+        response = self.ocp_client.resources.get(
+            api_version="template.openshift.io/v1", kind="Template"
+        ).get(namespace=self.infra_namespace, label_selector=_label_selector)
+        _LOGGER.debug(
+            "OpenShift response for getting provenance-checker template: %r",
```
Comment (fridex) [inline thoth/common/openshift.py]: This one is very specific for just provenance-checker, but the function used for other templates as well. What about logging just label selector which should be descriptive enough?

File: thoth/common/openshift.py [hunk 3]
```diff
@@ -687,9 +682,10 @@ def get_solver_names(self) -> list:
                 "Infra namespace is required in order to list solvers"
             )
 
-        response = self.ocp_client.resources.get(api_version="v1", kind="Template").get(
-            namespace=self.infra_namespace, label_selector="template=solver"
-        )
+        # TODO can we refactor this to use _get_template() ?
```
Comment (fridex) [inline thoth/common/openshift.py]: Maybe introduce `_get_templates()` instead and return template listing. Additional parameter (something like, `sanitize_only_one=True`) would check if there is just one. A check for != 0 would be performed always. I don't insist on this change as it's internal logic which would not introduce tech-dept elsewhere , just suggestion to next implementation.

**Comment by fridex** [comment]
> do we have to roll a new release now?

Yes, we will need to do that.

**Pull request merged**